pkg/pushsync/pushsync.go

```diff
@@ -139,9 +140,13 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 	ps.metrics.TotalReceived.Inc()
 
 	chunk := swarm.NewChunk(swarm.NewAddress(ch.Address), ch.Data)
-	if chunk, err = ps.validStamp(chunk, ch.Stamp); err != nil {
-		return fmt.Errorf("pushsync valid stamp: %w", err)
+	chunkAddress := chunk.Address()
```

Comment:
I guess setting the stamp here is not required if you use the new validStampBytes call which would do this.